pkg/app/app_reconcile.go

```diff
@@ -15,13 +15,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const appResourceType = "app"
+
 // Reconcile is not expected to be called concurrently
 func (a *App) Reconcile(force bool) (reconcile.Result, error) {
 	defer a.flushUpdateStatus("app reconciled")
 
 	var err error
 
-	a.appMetrics.InitMetrics(a.Name(), a.Namespace())
+	a.appMetrics.ReconcileCountMetrics.InitMetrics(appResourceType, a.Name(), a.Namespace())
```

Comment:
There is no a.Kind() available here

Reply:
The `kind` for an app shouldn't be a constant in a reconciler. It should live with the object model.

In this case, our `app` struct has the real app as a member, and we can call kind on it.

We should add another function to app to return the kind, rather than a constant.